CONTRIBUTING.md

```diff
@@ -44,6 +44,33 @@ make the code easier to navigate.
 
 *(TODO: Provide some notes on how to prepare a development environment which
 includes the needed dependencies.)*
+### Note
+This project requires libwpewebkit-1.0-dev, libwpe-1.0-dev, libwpebackend-fdo-1.0-dev 
+and GTK4 for the build/run process
+
+After cloning the project from [https://github.com/Igalia/cog] do the following 
+to build and get the project running
+-cd into project
+-create a build directory
+-cd into this directory. This is where your build files will be stored.
+-run cmake -DCOG_PLATFORM_X11=ON -DCOG_PLATFORM_GTK4=ON .. to generate the build files.
+-Then run the make command to generate the binaries
+You just finished building cog. To run it, run COG_MODULEDIR=$PWD/modules ./cog --platform=gtk4 [https://www.igalia.com/] 
+while in the build directory. If you are running Wayland, this will startup fine.
+
+### Building and Running Cog on Ubuntu 20.04
```

Comment:
I would omit this section, given Ubuntu 20.04 doesn't have Gtk4 or a new enough GLib to build Gtk4. Alternatively, I expect one could use a Flatpak sandbox, or just build without the GTK backend, but that's probably beyond the scope of this documentation.

Reply:
Is there some document I can read to help me better explain how someone can use the flatpak sandbox and also explain building without GTK backend?

Reply:
The Flatpak documentation is at https://docs.flatpak.org/en/latest/ and there's a simple guide on building a Flatpak package from a source tarball here: https://opensource.com/article/19/10/how-build-flatpak-packaging but I don't know how much effort that requires, my previous attempts at using Flatpak for development never went that well (it has been a long time since then though). As for building Cog without GTK, you just need to omit the `-DCOG_PLATFORM_GTK4=ON` option (or explicitly set it to `OFF`, though it's off by default last I checked).